hack/tests/e2e-helm.sh

```diff
@@ -1,169 +1,29 @@
 #!/usr/bin/env bash
 
-set -eux
+set -o errexit
```

Comment:
Revert this file back to the legacy Helm e2e test.

In a follow-up (when the legacy Helm scaffolding is removed) we can rename `e2e-helm-new-shell.sh` to `e2e-helm.sh`. And in a separate follow-up, we can port the actual tests from Shell-based tests to Go-based tests and update this file accordingly.